src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceOptionsPage.cs

```diff
@@ -25,13 +25,27 @@ protected override void OnActivate(CancelEventArgs e)
         {
             base.OnActivate(e);
             PackageSourcesControl.Font = VsShellUtilities.GetEnvironmentFont(this);
-            PackageSourcesControl.InitializeOnActivated();
+
+            // TODO: Do not wrap JTF.Run around brokered calls
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => await OnActivateAsync(e));
```

Comment:
In the meantime, please use the JTF.Run overload that provides its delegate with a `CancellationToken` and pass that into the async method so you can honor it.
This will give the customer the ability to cancel the operation and avoid an IDE hang if the operation takes too long.